ontopy/ontology.py

```diff
@@ -1051,23 +1051,38 @@ def is_defined(self, entity):
             entity = self.get_by_label(entity)
         return hasattr(entity, "equivalent_to") and bool(entity.equivalent_to)
 
-    def get_version(self, as_iri=False):
+    def get_version(self, as_iri=False) -> str:
         """Returns the version number of the ontology as inferred from the
-        owl:versionIRI tag.
+        owl:versionINFO tag or owl:versionIRI tag.
 
         If `as_iri` is True, the full versionIRI is returned.
         """
-        version_iri_storid = self.world._abbreviate(
-            "http://www.w3.org/2002/07/owl#versionIRI"
+        version_info_storid = self.world._abbreviate(
+            "http://www.w3.org/2002/07/owl#versionInfo"
         )
-        tokens = self.get_triples(s=self.storid, p=version_iri_storid)
+        tokens = self.get_triples(s=self.storid, p=version_info_storid)
+        version_iri_storid = None
         if not tokens:
-            raise TypeError(f"No versionIRI in Ontology {self.base_iri!r}")
+            version_iri_storid = self.world._abbreviate(
+                "http://www.w3.org/2002/07/owl#versionIRI"
+            )
+            tokens = self.get_triples(s=self.storid, p=version_iri_storid)
+            if not tokens:
+                raise TypeError(
+                    "No versionIRI or versionInfo "
+                    f"in Ontology {self.base_iri!r}"
+                )
         _, _, obj = tokens[0]
-        version_iri = self.world._unabbreviate(obj)
-        if as_iri:
-            return version_iri
-        return infer_version(self.base_iri, version_iri)
+        if isinstance(obj, int):
+            version_iri = self.world._unabbreviate(obj)
+        else:
+            version_iri = obj.strip('"').strip("'")
+
+        if as_iri and not version_iri_storid:
```

Comment:
The case `as_iri=True` and `version_iri_storid` is given is not considered. In that case we should simply return `versionIRI`. 

Note also the case where the ontology provides both `versionInfo` and `versionIRI`. If `as_iri` is true, we want to return `versionIRI`, but that is never fetched due to the if-statement on line 1065.

Reply:
if ´as_iri=True´ and ´version_iri_storid´ the last return statement return versionIRI, no?

if ´as_iri´ is true and version_info exists version_iri is constructed from base_iri. I agree this is not a good solution. This means that it should check for versionIRI alwasy is as_iri is true.

Reply:
Yes, you are right.